torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3621,8 +3582,48 @@ def shard_full_optim_state_dict(
             flattened parameters instead of unflattened parameters and
             restricted to only include this rank's part of the optimizer state.
         """
-        return _flatten_full_optim_state_dict(
-            full_optim_state_dict, model, True, optim_input,
+
+        return _flatten_optim_state_dict(
+            full_optim_state_dict, model=model, shard_state=True, optim_input=optim_input,
+        )[0]
+
+    @staticmethod
+    def flatten_sharded_optim_state_dict(
+        sharded_optim_state_dict: Dict[str, Any],
+        model: torch.nn.Module,
+        optim_input: Optional[
+            Union[
+                List[Dict[str, Any]], Iterable[torch.nn.Parameter],
+            ]
+        ] = None,
+    ) -> Dict[str, Any]:
+        """
+        The API is similar to :meth:``shard_full_optim_state_dict``. The only
+        difference is that the input ``full_optim_state_dict`` should be returned
```

Comment:
Not sure if I'm missing something, but the doc might not be correct? i.e. the input should be from `sharded_optim_state_dict`? let me know if I'm missing something